cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -208,6 +258,46 @@ macro(ov_arm_neon_fp16_optimization_flags flags)
     endif()
 endmacro()
 
+#
+# ov_arm_sve_optimization_flags(<output flags>)
+#
+macro(ov_arm_sve_optimization_flags flags)
+    if(OV_COMPILER_IS_INTEL_LLVM)
+        message(WARNING "Unsupported CXX compiler ${CMAKE_CXX_COMPILER_ID}")
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        # nothing to define; works out of box
+    elseif(ANDROID)
+        if(ANDROID_ABI STREQUAL "arm64-v8a")
+            set(${flags} -march=armv8-a+sve -Wno-unused-command-line-argument)
+        else()
+            message(WARNING "SVE is not supported on this Android ABI: ${ANDROID_ABI}")
+        endif()
+    else()
+        if(AARCH64)
+            ov_check_compiler_supports_sve(CXX "-march=armv8-a+sve")
+            ov_check_compiler_supports_sve(C "-march=armv8-a+sve")
+
+            if(CXX_SVE_FOUND AND C_SVE_FOUND)
+                set(${flags} -O2 -march=armv8-a+sve)
+            else()
+                set(${flags} -O2)
+            endif()
+
+            if(NOT CMAKE_CL_64)
+                list(APPEND ${flags} -ftree-vectorize)
+            endif()
+            # Check for SVE support
+            if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64" AND CXX_SVE_FOUND AND C_SVE_FOUND)
```

Comment:
do we need this condition? 

- `${flags}` contains SVE options if `if(CXX_SVE_FOUND AND C_SVE_FOUND)` is true (see above)
- we are already inside `if(AARCH64)`, so no needs to check target processor

Reply:
Makes sense. I've removed the redundant condition in the latest commit.